pinot-core/src/main/java/org/apache/pinot/core/data/table/Record.java

```diff
@@ -0,0 +1,42 @@
+package org.apache.pinot.core.data.table;
+
+import java.util.Arrays;
+import org.apache.pinot.common.utils.EqualityUtils;
+
+
+public class Record {
+  private Object[] _keys;
+  private Object[] _values;
+
+  public Record(Object[] keys, Object[] values) {
+    _keys = keys;
+    _values = values;
+  }
+
+  public Object[] getKeys() {
+    return _keys;
+  }
+
+  public Object[] getValues() {
+    return _values;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (EqualityUtils.isSameReference(this, o)) {
```

Comment:
it's ok to skip all these checks and just do Array.equals for performance reasons

Reply:
will do